### DISCUSSION TITLE
May I ask how to do node current limiting next?

### DISCUSSION THREAD

**Discussion by parlay96** [vercel/next.js#69479] [Help]
### Summary

May I ask how to do node current limiting next? Limit the number of requests or concurrency to ensure the normal operation of the system

**Comment by Juneezee** [comment]
If you are deploying to Vercel, you can follow the official guide here: https://vercel.com/guides/rate-limiting-edge-middleware-vercel-kv

If not, you can still implement rate limiting using Redis + Next.js middleware

**Comment by parlay96** [comment]
@Juneezee Thank you
